osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/FirmwareService.java

```diff
@@ -323,6 +283,12 @@ private void saveFirmwareVersionsReturnedFromDevice(
             firmwareVersion.getFirmwareModuleType().getDescription());
         continue;
       }
+
+      /*
+       * The DeviceFirmwareModule has a id, that prevents duplicates in the database
+       * The id of DeviceFirmwareModule is made with a combination of deviceId and firmwareModuleId
+       * There for we can just add a new object to the database here
+       */
```

Comment:
Database centric, as I was raise, I would  excpect a unique constraint in the database, but it works.

Reply:
Flyway script is added

Reply:
Couldn't this potentially result in `ERROR: duplicate key value violates unique constraint` exceptions when attempting to add a DeviceFirmwareModule with the same id?

Reply:
That is also what i expected, that's why i made a test, to test this case!
But as i described in the comments is this not the case